tools/cold-backup/run.sh

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+#set -x #debug
+__TMP="/tmp/strimzi-backup" && mkdir -p $__TMP
+__HOME="" && pushd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" >/dev/null \
+    && { __HOME=$PWD; popd >/dev/null; }
+__MPOD="maintenance"
+
+SOURCE_NS="kafka"
+TARGET_NS="kafka"
+CLUSTER_NAME="my-cluster"
+BACKUP_NAME="$SOURCE_NS-$(date +%Y%m%d)"
+BACKUP_HOME="/tmp"
+
+# PVC size must match with the corresponding CR value
+# PVC class must be empty when not available
+# specify number of JBOD volumes or zero
+ZOO_REPLICAS="3"
+ZOO_PVC_SIZE="1Gi"
+ZOO_PVC_CLASS=""
+KAFKA_REPLICAS="3"
+KAFKA_PVC_SIZE="10Gi"
+KAFKA_PVC_CLASS=""
+KAFKA_JBOD="2"
+
+# custom configmap names
+# i.e. external logging configuration (log4j)
+CUSTOM_CMS=(
+    "log4j-properties"
+    "custom-test"
+)
+
+# custom secret names
+# i.e. listener's certificate, image registry authentication
+CUSTOM_SECRETS=(
+    "external-listener-cert"
+    "registry-secret"
+    "custom-test"
+)
+
+__error() {
+    local message="$1"
+    echo "$message"
+    exit 1
+}
+
+__confirm() {
+    local ns="$1"
+    local proc="${FUNCNAME[1]}"
+    local user=$(oc whoami)
+    if [[ -n $ns && -n $user ]]; then
+        read -p "Confirm $proc of $ns namespace as $user (y/n) " reply
+        if [[ ! $reply =~ ^[Yy]$ ]]; then
+            exit 0
+        fi
+    else
+        __error "Context not ready"
+    fi
+}
+
+__export() {
+    local name="$1"
+    local label="${2-}"
+    if [[ -n $name ]]; then
+        local crs=$(oc get $name -o name -l "$label")
+        if [[ -n $crs ]]; then
+            echo "Exporting $name"
+            # delete runtime metadata expression
+            local exp="del(.metadata.namespace, .items[].metadata.namespace, \
+                .metadata.resourceVersion, .items[].metadata.resourceVersion, \
+                .metadata.selfLink, .items[].metadata.selfLink, \
+                .metadata.uid, .items[].metadata.uid, \
+                .status, .items[].status)"
```

Comment:
I thought there was a `kubectl get --export` which removed this stuff?

Reply:
This flag is deprecated so I created this alternative function.

Reply:
OK, that makes sense. But I wonder whether having an include list rather than an exclude list would be better, since we don't know what might get added in the future. But an include list might be tricker for `yq` to do.